File: python/view/tela_inicial.py
import tkinter as tk
from tkinter import ttk, messagebox
from typing import List, Callable, Optional

class TelaInicial:
    def __init__(self):
        self.window = tk.Toplevel()
        self.window.title("Análise do Campeonato Brasileiro")
        self.window.geometry("600x500")
        self.window.resizable(False, False)
        
        # Centralizar janela
        self.window.transient()
        self.window.grab_set()
        
        self._init_components()
        self._setup_layout()
        
        # Callbacks
        self._carregar_dados_callback = None
        self._ano_change_callback = None
        self._confirmar_callback = None
    
    def _init_components(self):
        """Inicializa os componentes da interface"""
        # Título
        self.lbl_titulo = tk.Label(
            self.window,
            text="Análise do Campeonato Brasileiro",
            font=("Arial", 24, "bold"),
            fg="#006400"
        )
        
        # Status
        self.lbl_status = tk.Label(
            self.window,
            text="Clique em 'Carregar Dados' para iniciar",
            font=("Arial", 12),
            fg="gray"
        )
        
        # Progress bar
        self.progress_bar = ttk.Progressbar(
            self.window,
            mode='determinate',
            length=400
        )
        
        # Botão carregar dados
        self.btn_carregar_dados = tk.Button(
            self.window,
            text="Carregar Dados",
            font=("Arial", 14, "bold"),
            bg="#006496",
            fg="white",
            width=15,
            height=1,
            command=self._on_carregar_dados
        )
        
        # ComboBox anos
        self.combo_ano = ttk.Combobox(
            self.window,
            font=("Arial", 14),
            state="disabled",
            width=15
        )
        self.combo_ano.bind("<<ComboboxSelected>>", self._on_ano_change)
        
        # ComboBox times
        self.combo_time = ttk.Combobox(
            self.window,
            font=("Arial", 14),
            state="disabled",
            width=25
        )
        
        # Botão confirmar
        self.btn_confirmar = tk.Button(
            self.window,
            text="Analisar Time",
            font=("Arial", 16, "bold"),
            bg="#007800",
            fg="white",
            width=15,
            height=1,
            state="disabled",
            command=self._on_confirmar
        )
    
    def _setup_layout(self):
        """Configura o layout da interface usando pack()"""
        # Título
        self.lbl_titulo.pack(pady=20)
        
        # Status
        self.lbl_status.pack(pady=10)
        
        # Botão carregar dados
        self.btn_carregar_dados.pack(pady=20)
        
        # Frame para seleções
        frame_selecoes = tk.Frame(self.window)
        frame_selecoes.pack(pady=30)
        
        # Ano - Layout horizontal
        frame_ano = tk.Frame(frame_selecoes)
        frame_ano.pack(pady=15, fill="x")
        
        tk.Label(frame_ano, text="Ano:", font=("Arial", 14), width=6, anchor="e").pack(side="left", padx=(0, 10))
        self.combo_ano.pack(side="left")
        
        # Time - Layout horizontal
        frame_time = tk.Frame(frame_selecoes)
        frame_time.pack(pady=15, fill="x")
        
        tk.Label(frame_time, text="Time:", font=("Arial", 14), width=6, anchor="e").pack(side="left", padx=(0, 10))
        self.combo_time.pack(side="left")
        
        # Botão confirmar
        self.btn_confirmar.pack(pady=30)
    
    def _on_carregar_dados(self):
        """Callback para carregar dados"""
        if self._carregar_dados_callback:
            self._carregar_dados_callback()
    
    def _on_ano_change(self, event):
        """Callback para mudança de ano"""
        if self._ano_change_callback:
            self._ano_change_callback()
    
    def _on_confirmar(self):
        """Callback para confirmar análise"""
        if self._confirmar_callback:
            self._confirmar_callback()
    
    def mostrar_carregamento(self, mostrar: bool):
        """Controla exibição do carregamento"""
        if mostrar:
            # Colocar progress bar antes do botão
            self.progress_bar.pack(pady=10)
            self.btn_carregar_dados.config(state="disabled")
            self.lbl_status.config(text="Carregando dados...")
        else:
            self.progress_bar.pack_forget()
            self.btn_carregar_dados.config(state="normal")
    
    def set_progresso_carregamento(self, progresso: int, mensagem: str):
        """Atualiza progresso do carregamento"""
        self.progress_bar['value'] = progresso
        self.lbl_status.config(text=mensagem)
        self.window.update()
    
    def dados_carregados(self, sucesso: bool, mensagem: str):
        """Callback para quando dados são carregados"""
        self.progress_bar.pack_forget()
        self.lbl_status.config(text=mensagem)
        
        if sucesso:
            self.btn_carregar_dados.pack_forget()
            self.combo_ano.config(state="readonly")
        else:
            self.btn_carregar_dados.config(state="normal")
    
    def set_anos(self, anos: List[int]):
        """Define anos disponíveis"""
        self.combo_ano['values'] = anos
        if anos:
            self.combo_ano.set(anos[0])
    
    def set_times(self, times: List[str]):
        """Define times disponíveis"""
        self.combo_time['values'] = times
        if times:
            self.combo_time.set(times[0])
            self.combo_time.config(state="readonly")
            self.btn_confirmar.config(state="normal")
        else:
            self.combo_time.config(state="disabled")
            self.btn_confirmar.config(state="disabled")
    
    def mostrar_erro(self, mensagem: str):
        """Mostra mensagem de erro"""
        messagebox.showerror("Erro", mensagem)
    
    def get_ano_selecionado(self) -> Optional[int]:
        """Retorna ano selecionado"""
        try:
            return int(self.combo_ano.get())
        except:
            return None
    
    def get_time_selecionado(self) -> str:
        """Retorna time selecionado"""
        return self.combo_time.get()
    
    def set_carregar_dados_listener(self, callback: Callable):
        """Define callback para carregar dados"""
        self._carregar_dados_callback = callback
    
    def set_ano_change_listener(self, callback: Callable):
        """Define callback para mudança de ano"""
        self._ano_change_callback = callback
    
    def set_confirmar_listener(self, callback: Callable):
        """Define callback para confirmar"""
        self._confirmar_callback = callback
    
    def show(self):
        """Mostra a janela"""
        self.window.deiconify()
    
    def hide(self):
        """Esconde a janela"""
        self.window.withdraw()

File: java/src/main/java/view/TelaInicial.java
package view;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionListener;
import java.util.List;

public class TelaInicial extends JFrame {
    private JComboBox<Integer> comboAno;
    private JComboBox<String> comboTime;
    private JButton btnConfirmar;
    private JButton btnCarregarDados;
    private JLabel lblTitulo;
    private JLabel lblStatus;
    private JProgressBar progressBar;
    
    public TelaInicial() {
        initComponents();
        setupLayout();
    }
    
    private void initComponents() {
        setTitle("Análise do Campeonato Brasileiro");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setSize(600, 400);
        setLocationRelativeTo(null);
        
        // Componentes
        lblTitulo = new JLabel("Análise do Campeonato Brasileiro", SwingConstants.CENTER);
        lblTitulo.setFont(new Font("Arial", Font.BOLD, 24));
        lblTitulo.setForeground(new Color(0, 100, 0));
        
        // Componentes novos para carregamento
        lblStatus = new JLabel("Clique em 'Carregar Dados' para iniciar", SwingConstants.CENTER);
        lblStatus.setFont(new Font("Arial", Font.PLAIN, 12));
        lblStatus.setForeground(Color.GRAY);
        
        progressBar = new JProgressBar();
        progressBar.setStringPainted(true);
        progressBar.setVisible(false);
        
        btnCarregarDados = new JButton("Carregar Dados");
        btnCarregarDados.setFont(new Font("Arial", Font.BOLD, 14));
        btnCarregarDados.setBackground(new Color(0, 100, 150));
        btnCarregarDados.setForeground(Color.WHITE);
        
        // ComboBox para anos (inicialmente vazio, será preenchido após carregar dados)
        comboAno = new JComboBox<>();
        comboAno.setFont(new Font("Arial", Font.PLAIN, 14));
        comboAno.setEnabled(false);
        
        // ComboBox para times (inicialmente vazio)
        comboTime = new JComboBox<>();
        comboTime.setFont(new Font("Arial", Font.PLAIN, 14));
        comboTime.setEnabled(false);
        
        btnConfirmar = new JButton("Analisar Time");
        btnConfirmar.setFont(new Font("Arial", Font.BOLD, 16));
        btnConfirmar.setBackground(new Color(0, 120, 0));
        btnConfirmar.setForeground(Color.WHITE);
        btnConfirmar.setEnabled(false);
    }
    
    private void setupLayout() {
        setLayout(new BorderLayout());
        
        // Painel principal
        JPanel painelPrincipal = new JPanel(new GridBagLayout());
        painelPrincipal.setBackground(Color.WHITE);
        GridBagConstraints gbc = new GridBagConstraints();
        
        // Título
        gbc.gridx = 0; gbc.gridy = 0;
        gbc.gridwidth = 2;
        gbc.insets = new Insets(20, 20, 20, 20);
        painelPrincipal.add(lblTitulo, gbc);
        
        // Status
        gbc.gridx = 0; gbc.gridy = 1;
        gbc.gridwidth = 2;
        gbc.insets = new Insets(5, 20, 5, 20);
        painelPrincipal.add(lblStatus, gbc);
        
        // Progress Bar
        gbc.gridx = 0; gbc.gridy = 2;
        gbc.gridwidth = 2;
        gbc.fill = GridBagConstraints.HORIZONTAL;
        gbc.insets = new Insets(5, 50, 15, 50);
        painelPrincipal.add(progressBar, gbc);
        
        // Botão Carregar Dados
        gbc.gridx = 0; gbc.gridy = 3;
        gbc.gridwidth = 2;
        gbc.fill = GridBagConstraints.NONE;
        gbc.insets = new Insets(10, 20, 20, 20);
        painelPrincipal.add(btnCarregarDados, gbc);
        
        // Label e ComboBox Ano
        gbc.gridx = 0; gbc.gridy = 4;
        gbc.gridwidth = 1;
        gbc.insets = new Insets(10, 20, 10, 10);
        gbc.anchor = GridBagConstraints.EAST;
        painelPrincipal.add(new JLabel("Selecione o ano:"), gbc);
        
        gbc.gridx = 1; gbc.gridy = 4;
        gbc.anchor = GridBagConstraints.WEST;
        gbc.insets = new Insets(10, 10, 10, 20);
        painelPrincipal.add(comboAno, gbc);
        
        // Label e ComboBox Time
        gbc.gridx = 0; gbc.gridy = 5;
        gbc.anchor = GridBagConstraints.EAST;
        gbc.insets = new Insets(10, 20, 10, 10);
        painelPrincipal.add(new JLabel("Selecione o time:"), gbc);
        
        gbc.gridx = 1; gbc.gridy = 5;
        gbc.anchor = GridBagConstraints.WEST;
        gbc.insets = new Insets(10, 10, 10, 20);
        painelPrincipal.add(comboTime, gbc);
        
        // Botão Confirmar
        gbc.gridx = 0; gbc.gridy = 6;
        gbc.gridwidth = 2;
        gbc.insets = new Insets(30, 20, 20, 20);
        gbc.anchor = GridBagConstraints.CENTER;
        painelPrincipal.add(btnConfirmar, gbc);
        
        add(painelPrincipal, BorderLayout.CENTER);
    }
    
    // Métodos para controlar interface
    public void mostrarCarregamento(boolean mostrar) {
        progressBar.setVisible(mostrar);
        btnCarregarDados.setEnabled(!mostrar);
        if (mostrar) {
            lblStatus.setText("Carregando dados...");
        }
    }
    
    public void setProgressoCarregamento(int progresso, String mensagem) {
        progressBar.setValue(progresso);
        lblStatus.setText(mensagem);
    }
    
    public void dadosCarregados(boolean sucesso, String mensagem) {
        progressBar.setVisible(false);
        lblStatus.setText(mensagem);
        if (sucesso) {
            btnCarregarDados.setVisible(false);
            comboAno.setEnabled(true);
        } else {
            btnCarregarDados.setEnabled(true);
        }
    }
    
    // Métodos para configurar listeners
    public void setCarregarDadosListener(ActionListener listener) {
        btnCarregarDados.addActionListener(listener);
    }
    
    public void setAnoChangeListener(ActionListener listener) {
        comboAno.addActionListener(listener);
    }
    
    public void setConfirmarListener(ActionListener listener) {
        btnConfirmar.addActionListener(listener);
    }
    
    // Métodos para atualizar interface
    public void setAnos(List<Integer> anos) {
        comboAno.removeAllItems();
        for (Integer ano : anos) {
            comboAno.addItem(ano);
        }
    }
    
    public void setTimes(List<String> times) {
        comboTime.removeAllItems();
        for (String time : times) {
            comboTime.addItem(time);
        }
        comboTime.setEnabled(!times.isEmpty());
        btnConfirmar.setEnabled(!times.isEmpty());
    }
    
    public void mostrarErro(String mensagem) {
        JOptionPane.showMessageDialog(this, mensagem, "Erro", JOptionPane.ERROR_MESSAGE);
    }
    
    // Getters
    public Integer getAnoSelecionado() {
        return (Integer) comboAno.getSelectedItem();
    }
    
    public String getTimeSelecionado() {
        return (String) comboTime.getSelectedItem();
    }
}